### ISSUE TITLE
no gamma for feature (mcowner:9) !

### ISSUE THREAD

**Issue by pnprog** [pasky/pachi#108]
Hi,

Recently, I started to notice this issue, and it's happens quite a lot during game analysis (until now, I had not really tried to catch the **stderr** message, thinking it was more a GRP issue, so I did not know what was causing this issue). What does that means?

I encounter that when analysing games moves by moves, after each move, I **clear_board**, then replay the entire game before asking for a **genmove**. Then sometime (in the example below, after 200 moves) Pachi just crashes :(

```
IN: clear_board
IN: komi 0.5
IN: set_free_handicap D4 Q4 K10 D16 Q16
Move:   5  Komi: 0.5  Handicap: 5  Captures B: 0 W: 0
A B C D E F G H J K L M N O P Q R S T
+---------------------------------------+
19 | . . . . . . . . . . . . . . . . . . . |
18 | . . . . . . . . . . . . . . . . . . . |
17 | . . . . . . . . . . . . . . . . . . . |
16 | . . . X . . . . . . . . . . . X). . . |
15 | . . . . . . . . . . . . . . . . . . . |
14 | . . . . . . . . . . . . . . . . . . . |
13 | . . . . . . . . . . . . . . . . . . . |
12 | . . . . . . . . . . . . . . . . . . . |
11 | . . . . . . . . . . . . . . . . . . . |
10 | . . . . . . . . . X . . . . . . . . . |
9 | . . . . . . . . . . . . . . . . . . . |
8 | . . . . . . . . . . . . . . . . . . . |
7 | . . . . . . . . . . . . . . . . . . . |
6 | . . . . . . . . . . . . . . . . . . . |
5 | . . . . . . . . . . . . . . . . . . . |
4 | . . . X . . . . . . . . . . . X . . . |
3 | . . . . . . . . . . . . . . . . . . . |
2 | . . . . . . . . . . . . . . . . . . . |
1 | . . . . . . . . . . . . . . . . . . . |
+---------------------------------------+

IN: play white O3
IN: play black R6
IN: play white R10
IN: play black R14
IN: play white F3
IN: play black C6
IN: play white R3
IN: play black Q3
IN: play white R4
IN: play black R5
IN: play white Q2
IN: play black P2
IN: play white Q1
IN: play black P3
IN: play white O2
IN: play black P1
IN: play white S2
IN: play black O4
IN: play white N4
IN: play black O5
IN: play white N5
IN: play black O6
IN: play white L3
IN: play black N6
IN: play white M6
IN: play black R8
IN: play white R9
IN: play black N8
IN: play white Q8
IN: play black R7
IN: play white P7
IN: play black P6
IN: play white P9
IN: play black R12
IN: play white S11
IN: play black S12
IN: play white S8
IN: play black S7
IN: play white T8
IN: play black T7
IN: play white S10
IN: play black P11
IN: play white O10
IN: play black O11
IN: play white N10
IN: play black M7
IN: play white L6
IN: play black L7
IN: play white Q13
IN: play black Q14
IN: play white R17
IN: play black Q17
IN: play white R18
IN: play black R16
IN: play white S16
IN: play black S15
IN: play white T16
IN: play black T15
IN: play white S18
IN: play black Q18
IN: play white T17
IN: play black M17
IN: play white P18
IN: play black Q19
IN: play white N17
IN: play black N16
IN: play white F17
IN: play black J17
IN: play white P13
IN: play black O14
IN: play white O13
IN: play black N11
IN: play white P14
IN: play black P15
IN: play white M9
IN: play black L9
IN: play white N14
IN: play black O15
IN: play white M13
IN: play black M10
IN: play white M11
IN: play black L10
IN: play white Q11
IN: play black M12
IN: play white Q12
IN: play black L11
IN: play white N12
IN: play black L13
IN: play white L14
IN: play black M15
IN: play white M14
IN: play black K13
IN: play white K15
IN: play black M16
IN: play white K17
IN: play black K18
IN: play white H17
IN: play black K16
IN: play white J16
IN: play black L17
IN: play white H15
IN: play black K14
IN: play white L15
IN: play black F15
IN: play white C14
IN: play black C17
IN: play white C11
IN: play black C9
IN: play white F13
IN: play black E11
IN: play white B16
IN: play black B17
IN: play white E14
IN: play black D12
IN: play white C12
IN: play black D13
IN: play white D14
IN: play black E17
IN: play white F16
IN: play black F18
IN: play white E15
IN: play black G17
IN: play white G16
IN: play black H18
IN: play white G18
IN: play black E18
IN: play white D10
IN: play black E10
IN: play white D9
IN: play black C8
IN: play white D8
IN: play black D7
IN: play white F8
IN: play black G10
IN: play white H9
IN: play black H10
IN: play white D2
IN: play black C2
IN: play white D3
IN: play black C3
IN: play white E4
IN: play black D5
IN: play white E7
IN: play black E6
IN: play white F6
IN: play black K6
IN: play white K5
IN: play black J6
IN: play white J5
IN: play black H14
IN: play white G14
IN: play black H12
IN: play white H13
IN: play black J14
IN: play white G12
IN: play black J15
IN: play white P10
IN: play black O8
IN: play white H16
IN: play black J12
IN: play white H6
IN: play black J9
IN: play white J7
IN: play black K7
IN: play white J8
IN: play black G9
IN: play white H8
IN: play black G8
IN: play white H7
IN: play black G7
IN: play white F7
IN: play black G6
IN: play white G5
IN: play black F9
IN: play white E5
IN: play black E9
IN: play white E8
IN: play black C10
IN: play white D11
IN: play black E12
IN: play white K8
IN: play black L8
IN: play white H19
IN: play black J18
IN: play white B10
IN: play black C15
IN: play white B15
IN: play black E16
IN: play white D15
IN: play black D18
IN: play white B7
IN: play black C7
IN: play white A8
IN: play black B6
IN: play white B9
IN: play black T4
IN: play white T3
IN: play black S4
IN: play white R1
IN: genmove black
Move:  206  Komi: 0.5  Handicap: 5  Captures B: 2 W: 1
 A B C D E F G H J K L M N O P Q R S T        A B C D E F G H J K L M N O P Q R S T
   +---------------------------------------+    +---------------------------------------+
19 | . . . . . . . O . . . . . . . X . . . | 19 | : : : : : : : : : : : : : : : : : : : |
18 | . . . X X X O X X X . . . . O X O O . | 18 | : : : : : : : : : : : : : : : : : : : |
17 | . X X . X O . O X . X X O . . X O . O | 17 | : : : : : : : : : : : : : : : : : : : |
16 | . O . X X O O O O X . X X . . X X O O | 16 | : : : : : : : : : : : : : : : : : : : |
15 | . O X O O X . O X O O X . X X . . X X | 15 | : : : : : : : : : : : : : : : : : : : |
14 | . . O O O . O X X X O O O X O X X . . | 14 | : : : : : : : : : : : : : : : : : : : |
13 | . . . X . O . O . X X O . O O O . . . | 13 | : : : : : : : : : : : : : : : : : : : |
12 | . . O X X . O X X . . X O . . O X X . | 12 | : : : : : : : : : : : : : : : : : : : |
11 | . . O O X . . . . . X . X X X O . O . | 11 | : : : : : : : : : : : : : : : : : : : |
10 | . O X O X . X X . X X X O O O . O O . | 10 | : : : : : : : : : : : : : : : : : : : |
 9 | . O X O X X X O X . X O . . O . O . . |  9 | : : : : : : : : : : : : : : : : : : : |
 8 | O . X O O O X O O O X . X X . O X O O |  8 | : : : : : : : : : : : : : : : : : : : |
 7 | . O X X O O X O O X X X . . O . X X X |  7 | : : : : : : : : : : : : : : : : : : : |
 6 | . X X . X O X O X X O O X X X . X . . |  6 | : : : : : : : : : : : : : : : : : : : |
 5 | . . . X O . O . O O . . O X . . X X . |  5 | : : : : : : : : : : : : : : : : : : : |
 4 | . . . X O . . . . . . . O X . X O X X |  4 | : : : : : : : : : : : : : : : : : : : |
 3 | . . X O . O . . . . O . . O X X O O O |  3 | : : : : : : : : : : : : : : : : : : : |
 2 | . . X O . . . . . . . . . O X O . O . |  2 | : : : : : : : : : : : : : : : : : : : |
 1 | . . . . . . . . . . . . . . X O O). . |  1 | : : : : : : : : : : : : : : : : : : : |
   +---------------------------------------+    +---------------------------------------+
mcowner 0.08s
dcnn in 0.27s
no gamma for feature (mcowner:9) !
```

**Comment by pnprog** [comment]
Also,

+ I think this happens only with handicap games
+ I am not using the latest release, but compiled version from last September I guess. I can try again with latest version if it could have been solved since then

**Comment by lemonsqueeze** [comment]
Yes, it was there for a long time, happened rarely with multithreading
(possibly more if you have lots of threads).
Fixed in PR #99 so if you use current master it shouldn't happen anymore.

**Comment by pnprog** [comment]
Thanks, I just recompiled Pachi, and it seems to have solve the issue.
